{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Upload Single Cell data to Broad Single Cell Portal\n",
    "L.Richards  \n",
    "April 2020  \n",
    "\n",
    "----\n",
    "\n",
    "Need to upload clustering, metadata and normalized expression matrix files to Broad Single Cell portal.Uploading in-house scRNA-seq data of MB cell lines only. In the same configuration as Figure 2A of Branavans paper.\n",
    "\n",
    ">https://singlecell.broadinstitute.org/single_cell/study/SCP840: this link should be added to data availability section of manuscript\n",
    ">  sheilasinghlab@gmail.com & chitrave@gmail.com have login and edit access to account\n",
    "\n",
    "**Currently the study is private, but upon publication this accession should be made public**\n",
    "\n",
    "/mnt/work1/users/pughlab/projects/SinghLab_scMedullo/Brananvan_Wnt/Broad_scRNA_Portal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "library(\"data.table\")\n",
    "library(\"dplyr\")\n",
    "library(\"taRifx\")\n",
    "library(\"Matrix\")\n",
    "suppressMessages(library(\"Seurat\", \n",
    "                         lib =\"/mnt/work1/users/pughlab/projects/BTSCs_scRNAseq/Manuscript/lib/\" ))\n",
    "\n",
    "\n",
    "\n",
    "SeuratPath <- \"~/pughlab/projects/SinghLab_scMedullo/Brananvan_Wnt/AddModuleScore/WntMedullo_newQC_scoring.RData\"\n",
    "sample <- \"Singh_BTIC_Medulloblastoma\"\n",
    "\n",
    "\n",
    "\n",
    "################################################\n",
    "\n",
    "print(\"\")\n",
    "print(\"-------\")\n",
    "print(sample)\n",
    "start.time <- Sys.time()\n",
    "print(start.time)\n",
    "print(\"-------\")\n",
    "\n",
    "################################################\n",
    "print(\"\")\n",
    "print(\"\")\n",
    "print(\"Loading Seurat Object\")\n",
    "print(Sys.time())\n",
    "print(SeuratPath)\n",
    "\n",
    "loadRData <- function(fileName){\n",
    "#loads an RData file, and returns it\n",
    "    load(fileName)\n",
    "    get(ls()[ls() != \"fileName\"])\n",
    "}\n",
    "      \n",
    "dat <- loadRData(SeuratPath)\n",
    "print(dim(dat@data))\n",
    "\n",
    "################################################\n",
    "print(\"\")\n",
    "print(\"\")\n",
    "print(\"Extract Normalized Gene Expression File (txt.gz)\")\n",
    "print(Sys.time())\n",
    "\n",
    "norm.exp <- data.frame(as.matrix(dat@data))\n",
    "norm.exp <- tibble::rownames_to_column(norm.exp, \"GENE\")\n",
    "\n",
    "\n",
    "print(norm.exp[1:5, 1:5])\n",
    "\n",
    "file1 <- paste0(sample, \"_NormalizedExpression.txt.gz\")\n",
    "print(file1)\n",
    "      \n",
    "write.table(norm.exp, \n",
    "          file=gzfile(file1),\n",
    "          row.names = FALSE,\n",
    "          quote = FALSE,\n",
    "            sep = \"\\t\"\n",
    "          )      \n",
    "\n",
    "\n",
    "################################################\n",
    "print(\"\")\n",
    "print(\"\")\n",
    "print(\"Extract Meta Data (.txt)\")\n",
    "print(Sys.time())\n",
    "\n",
    "meta <- data.frame(dat@meta.data)\n",
    "meta <- remove.factors(meta)\n",
    "\n",
    "include <- c(\"nGene\",\n",
    "             \"nUMI\",\n",
    "             \"percent.mito\",\n",
    "             \"Num.Reads\",\n",
    "             \"Sample\",\n",
    "             \"Subgroup\",\n",
    "             \"Stage\",\n",
    "             colnames(dat@meta.data)[grep(\"GSVA.scaled\", colnames(dat@meta.data))]\n",
    "            )\n",
    "meta <- meta[,include]\n",
    "colnames(meta) <- gsub(\"GSVA.scaled\", \"GSVA\", colnames(meta))\n",
    "meta$nGene <- as.numeric(meta$nGene)\n",
    "meta$nUMI <- as.numeric(meta$nUMI)\n",
    "meta$nUMI <- as.numeric(meta$Num.Reads)\n",
    "\n",
    "classes <- as.vector(unlist(lapply(meta, class)))\n",
    "TYPE <- c(\"TYPE\", ifelse(classes == \"numeric\", \"numeric\", \"group\"))\n",
    "meta <- data.frame(setDT(meta, keep.rownames = \"NAME\"))\n",
    "meta <- rbind(TYPE, meta)\n",
    "      \n",
    "file2 <- paste0(sample, \"_MetaData.txt\")\n",
    "print(file2)\n",
    "\n",
    "write.table(meta, \n",
    "          file=file2,\n",
    "          row.names = FALSE,\n",
    "          quote = FALSE,\n",
    "          sep = \"\\t\",\n",
    "          col.names = TRUE\n",
    "          ) \n",
    "\n",
    "\n",
    "################################################\n",
    "print(\"\")\n",
    "print(\"\")\n",
    "print(\"Extract tSNE Coordinates (.txt)\")\n",
    "print(Sys.time())\n",
    "\n",
    "cluster.coords <- as.data.frame(dat@dr$tsne@cell.embeddings)[, c(1:2)]\n",
    "colnames(cluster.coords) <- c(\"X\", \"Y\")\n",
    "cluster.coords <- remove.factors(cluster.coords)\n",
    "\n",
    "classes <- as.vector(unlist(lapply(cluster.coords, class)))\n",
    "TYPE <- c(\"TYPE\", ifelse(classes == \"numeric\", \"numeric\"))\n",
    "cluster.coords <- data.frame(setDT(cluster.coords, keep.rownames = \"NAME\"))\n",
    "cluster.coords  <- rbind(TYPE, cluster.coords)\n",
    "\n",
    "##subset out some meta data to include as labels\n",
    "cluster.coords <- cbind(cluster.coords, meta[ ,c(\"Sample\",\n",
    "                                                 \"Subgroup\",\n",
    "                                                 \"Stage\"\n",
    "                                                )] )\n",
    "\n",
    "file3 <- paste0(sample, \"_tSNE_ClusterCoordinates.txt\")\n",
    "print(file3)\n",
    "write.table(cluster.coords, \n",
    "          file=file3,\n",
    "          row.names = FALSE,\n",
    "          quote = FALSE,\n",
    "           sep = \"\\t\",\n",
    "          col.names = TRUE\n",
    "          ) \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
